stan/math/prim/prob/discrete_range_lpmf.hpp

```diff
@@ -0,0 +1,96 @@
+#ifndef STAN_MATH_PRIM_PROB_DISCRETE_RANGE_LPMF_HPP
+#define STAN_MATH_PRIM_PROB_DISCRETE_RANGE_LPMF_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/fun/constants.hpp>
+#include <stan/math/prim/fun/inv.hpp>
+#include <stan/math/prim/fun/log.hpp>
+#include <stan/math/prim/fun/max_size.hpp>
+#include <stan/math/prim/fun/size_zero.hpp>
+#include <stan/math/prim/fun/value_of.hpp>
+#include <cmath>
+
+namespace stan {
+namespace math {
+
+/** \ingroup prob_dists
+ * Return the log PMF of a discrete range for the given y, lower and upper
+ * bound.
+ *
+ * `lower` and `upper` can each be a scalar or a one-dimensional container.
+ * Any non-scalar inputs must be the same size.
+ *
+ * @tparam T_y type of scalar, either int or std::vector<int>
+ * @tparam T_lower type of lower bound, either int or std::vector<int>
+ * @tparam T_upper type of upper bound, either int or std::vector<int>
+ *
+ * @param y random variable
+ * @param lower lower bound
+ * @param upper upper bound
+ * @return Log probability. If containers are supplied, returns the log sum
+ * of the probabilities.
+ * @throw std::domain_error if upper is smaller than lower.
+ * @throw std::invalid_argument if non-scalar arguments are of different
+ * sizes.
+ */
+template <bool propto, typename T_y, typename T_lower, typename T_upper>
+double discrete_range_lpmf(const T_y& y, const T_lower& lower,
+                           const T_upper& upper) {
+  static const char* function = "discrete_range_lpmf";
+
```

Comment:
[optional]
Vertical whitespace is precious!  I'd get rid of all these extra spaces to allow myself to view more of the function at once.

Reply:
I cut a couple of empty lines. I still prefer keeping logically separate block also separate visually.

Reply:
That's why I made it optional.  No promises I won't remove vertical space if I ever touch the files in the future :-)